### ISSUE TITLE
Definition Provider from MPLS broken

### ISSUE THREAD

**Issue by wookayin** [neoclide/coc-python#54]
`CocAction('jumpDefinition')` always cannot find a definition when MPLS is used. Jedi works `"python.jediEnabled": true`, so I suppose this is a bug from language server (or its coc-python bridge).

When `jumpDefinition` is triggered (on user packages and *standard* packages), I get either:

- [coc.nvim] Definition not found
- [coc.nvim] Definition provider not found for current document

Any idea about how to debug this? `:CocInfo` doesn't give anything useful.

Version information:

```
vim version: NVIM v0.4.0-914-g4841c46e3
node version: v12.4.0
coc.nvim version: 0.0.71
term: xterm-256color
platform: linux
```

```
coc-python 1.2.1 ~/.config/coc/extensions/node_modules/coc-python
coc-python-data: languageServer.0.2.96
```

**Comment by chemzqm** [comment]
https://github.com/neoclide/coc.nvim/wiki/Debug-language-server#using-output-channel

**Comment by wookayin** [comment]
Thanks, and sorry about not having carefully gone through the FAQ.

Seems a bug of MPLS.

```
[Trace - 4:27:29 PM] Received response 'textDocument/completion - (19)' in 2ms. Request failed: Object reference not set to an instance of an object. (-32000).
Error data: "   at Microsoft.Python.LanguageServer.Implementation.LanguageServer.Completion(JToken token, CancellationToken cancellationToken) in E:\\A\\_work\\132\\s\\src\\LanguageS
erver\\Impl\\LanguageServer.cs:line 212"


[Error  - 4:27:29 PM] Request textDocument/completion failed.
  Message: Object reference not set to an instance of an object.
  Code: -32000·
   at Microsoft.Python.LanguageServer.Implementation.LanguageServer.Completion(JToken token, CancellationToken cancellationToken) in E:\A\_work\132\s\src\LanguageServer\Impl\Language
Server.cs:line 212
```

```
[Trace - 4:33:09 PM] Sending request 'textDocument/definition - (1)'.
Params: {
    "textDocument": {
        "uri": "file:///(PATH HIDDEN)/a.py"
    },
    "position": {
        "line": 11,
        "character": 0
    }
}

[Trace - 4:33:09 PM] Received notification 'telemetry/event'.
Params: {
    "EventName": "python_language_server/rpc.request",
    "Properties": {
        "method": "textDocument/definition",
        "plsVersion": "0.2.96.0"
    },
    "Measurements": {
        "elapsedMs": 223.4839
    }
}
```

Should I fire a bug on microsoft/python-language-server?

**Comment by chemzqm** [comment]
> Should I fire a bug on microsoft/python-language-server?

It's up to you. Maybe try with daily channel of MPLS to see if it's fixed.

**Comment by wookayin** [comment]
I see. I've upgraded the python language server to latest (0.3.20) and the bug is now gone.

* `:CocCommand python.upgradePythonLanguageServer`

I thought it was the latest when I was writing this issue (I had removed coc-python-data and reinstall `:CocInstall coc-python` but it still downloaded the same version 0.2.96, which is why I thought so) but explicitly requesting an upgrade gave me a latest version.


Plus, just reference for who might find this issue later on (for debugging language servers):

- `:CocCommand python.viewOutput` is pretty useful for debugging.
